content/src/logic/snapshot-cleaner.ts

```diff
@@ -0,0 +1,95 @@
+import { exec } from 'child_process'
+import { ReadStream } from 'fs'
+import PQueue from 'p-queue'
+import path from 'path'
+import { promisify } from 'util'
+import { streamToBuffer } from '../ports/contentStorage/contentStorage'
+import { FSComponent } from '../ports/fs'
+import { AppComponents } from '../types'
+import { filterContentFileHashes } from './database-queries/snapshot-cleaner-queries'
+
+export async function cleanSnapshots(
+  components: Pick<AppComponents, 'logs' | 'gzipCompressor' | 'database'> & {
+    fs: Pick<FSComponent, 'unlink' | 'createReadStream'>
+  },
+  storageDirectory: string,
+  minimumSnapshotSizeInBytes: number
+): Promise<void> {
+  return cleanSnapshotsWithExec(components, promisify(exec), storageDirectory, minimumSnapshotSizeInBytes)
+}
+
+// This function that takes exectueCommand that is for injecting a mock and being able to unit test.
+// Didn't found a way to mock exec/promisify functions.
+export async function cleanSnapshotsWithExec(
+  components: Pick<AppComponents, 'logs' | 'gzipCompressor' | 'database'> & {
+    fs: Pick<FSComponent, 'unlink' | 'createReadStream'>
+  },
+  executeCommand: (command: string) => Promise<{ stdout: string; stderr: string }>,
+  storageDirectory: string,
+  minimumSnapshotSizeInBytes: number
+): Promise<void> {
+  const logger = components.logs.getLogger('Snapshot Cleaner')
+
+  const isModernSnapshotHeader = (header: string): boolean => header.startsWith('### Decentraland json snapshot')
+  const isLegacySnapshotHeader = (header: string): boolean => header.match(/\[\[\"Qm.+?(?=\",\[\")/) !== null
```

Comment:
Can this two conditions be removed and only remove big files that are not part of any active deployment? This is tooooo specific and I see more value on having this same code but with fewer conditions.

Anyways, as part of GC, we want to delete non-important files, regardless if they are snapshots or not. And that's already being done by `filterContentFileHashes`

Reply:
I think that would simplify _a lot_ the proposed implementation without losing any value

Reply:
Yes, it would be simpler but less secure in some way. I still don't have the expertise on the content server to evaluate the security of that decision, but i believe in you :)

Reply:
I will defer to the team here for a final decision. From my understanding, as long as the process runs _before_ you start the server component or bind the router, then it should be safe. Because there are no other parts of the code touching these files and the server is definitely not serving the files to anyone

Reply:
I prefer to delete first only the snapshots files, as there are many and they take up lots of space. After that, we can analyze if there are still big files unreferenced and what to do with them. In the case we need to delete them as well, it'd be as easy as removing these checks on the next release.